uint32 type
string log_info
string source
